src/test/integration/integration-test.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import json
+import unittest
+import yaml
+import utils
+
+
+class integration_tests(unittest.TestCase):
+    @classmethod
+    def setup(cls, config_path, openapi_path):
+        with open(config_path) as config_file:
+            config = json.load(config_file)
+            cls.base_url = utils.setup_base_url(config)
+            cls.session = utils.setup_session(config)
+            cls.test_cases = config['test_cases']
+
+        with open(openapi_path) as openapi_file:
+            cls.openapi = yaml.load(openapi_file)
+
+    @classmethod
+    def cleanup(cls):
+        cls.session.close()
+
+    # Test case: GET /spendingauthority with valid authority onids
+    def test_get_authority_onids(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['valid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = utils.get_resource_schema(
+                self, 'SpendingAuthorityResource'
+            )
+            utils.check_schema(self, response, spending_schema)
+            response_data = response.json()['data']
+            actual_onid = response_data['id']
+            self.assertEqual(actual_onid.lower(), onid.lower())
+
+    # Test case: GET /spendingauthority with invalid authority onids
+    def test_get_non_authority_onids(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['invalid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = []
+            self.assertEqual(response.json()['data'], spending_schema)
+
+    # Test case: GET /spendingauthority with bad request
+    def test_error_response(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['bad_request']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 400, params=params)
+            error_schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, error_schema)
+
+    # Test case: GET /spendingauthority spending limits
+    def test_spending_limits(self, endpoint='/spendingauthority'):
```

Comment:
Is there any specific reason you wanna separate this test case with `test_get_authority_onids`? Both of these test cases are expecting 200 response and querying with the same list of valid onid ids.

Reply:
Yes, the first test `test_get_authority_onids`  is testing that the onid have spending authority, `test_spending_limits` is testing that the spending limits and indexes are correct. Separating them would make it easier to debug and detect the source of the error.

Reply:
@MohammedAlawami Ok, sounds fair. That way, instead of sending the same requests with the same `valid_authority_onids` for both testing cases, can you figure out a way to try only send one request for both tests? We don't want to send exactly the same request twice in the integration test.